### PR TITLE
AVRO-3264: Improvements to the project section of the Avro website

### PR DESCRIPTION
**Pull request by nishchaldv**
## What is the purpose of the change
This pull request improves the Avro website's project section fixing AVRO-3264.

## Verifying this change
This change is a trivial rework without any test coverage.

## Documentation

This pull request improves the Avro website's project section in several ways. First, it adds papers and articles that can help users better understand Avro. Second, it includes onboarding guidelines for committers, contributors, and PMC members, which can assist them in understanding their workflows. Finally, it improves the 'How to Contribute' page by providing explanations on how to make code changes by cloning the project locally and running the avro-website inside a Docker container.

- Does this pull request introduce a new feature? 
   No

### CODE REVIEW

File: doc/content/en/project/Committer onboarding guide/_index.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ specific language governing permissions and limitations
+ under the License.
+
+--> 
+** **
+For you, the new committer:
+
+1. File your ICLA and send it to secretary@apache.org
+2. Log in to https://whimsy.apache.org; that will confirm a working ASF account
+3. you can edit email routing for the account, and add other emails that you own
```
Comment (martin-g) [inline doc/content/en/project/Committer onboarding guide/_index.md:32]:
```suggestion
3. You can edit email routing for the account, and add other emails that you own
```

File: doc/content/en/project/Committer onboarding guide/_index.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+4. you can directly edit mailing list subscriptions (for example, you might switch them to your ASF account - you can still post from any of your registered emails)
```
Comment (martin-g) [inline doc/content/en/project/Committer onboarding guide/_index.md:33]:
```suggestion
4. You can directly edit mailing list subscriptions (for example, you might switch them to your ASF account - you can still post from any of your registered emails)
```

File: doc/content/en/project/How to contribute/_index.md [hunk 3]
```diff
@@ -276,7 +275,114 @@ Please refrain from editing descriptions and comments if possible, as edits spam
 
 Contributors should join the Avro mailing lists. In particular, the commit list (to see changes as they are made), the dev list (to join discussions of changes) and the user list (to help others).
 
+## Workflow
+
+Building and running the site locally requires a recent extended version of Hugo. Install Hugo for your environment. Once you've made your working copy of the site repo, from the repo root folder, run:
```
Comment (martin-g) [inline doc/content/en/project/How to contribute/_index.md:280]: IMO it would be useful to have a link to Hugo download page

**Review by martin-g — APPROVED** [review]

**Review by RyanSkraba — APPROVED** [review]
Hey, thanks for the contribution -- and my excuses for taking a while to review and merge this!  I'm having an issue with the build process when we have more than one release.  For the moment, 1.11.1 is the only Avro release that actually has the hugo-generated pages.  

@martin-g has been looking at being able to automatically deploy the project pages, and I've been looking at how we can possibly deploy the release pages independently from project pages...  I'm not sure when we can actually get your changes reflected on the Avro website, but I'm glad to have them captured in the repository now!

**Comment by martin-g** [comment]
I can revive my work on the automatic website deployment.
The stopper was that I had/have no idea how to support the separate SDKs but since this is also not done I will just push it.

https://github.com/apache/arrow-datafusion/pull/5670 - here is a PR that I just finished for Apache Arrow Datafusion.

**Comment by RyanSkraba** [comment]
I think it gets worse than separate SDKs!  I don't see how we can have automatic deployment of some of the pages (which are "release" pages), as opposed to project pages... The spec, for example, is associated with a release (as are the JavaDocs, other API docs and examples).

If you can get the project pages auto-deployed, however, we can worry about those other pages.  I'm currently testing how we can separate them by copying how Flink does a separate [nightlies](https://nightlies.apache.org/flink/flink-docs-release-1.16/) page per-branch.

(I like to copy whatever a bigger project does..!)

**Comment by RyanSkraba** [comment]
Also if you can get your auto-deploy to work, we can restore the "Edit this page" buttons! \o/

**Comment by nishchaldv** [comment]
Thank you, @RyanSkraba and @martin-g, for reviewing my work.

**Pull request merged**